Volume/Views/PublicationDetail/PublicationDetailHeader.swift

```diff
@@ -21,39 +21,33 @@ struct PublicationDetailHeader: View {
     private var shoutouts: Int {
         max(publication.shoutouts, userData.shoutoutsCache[publication.id, default: 0])
     }
+    
+    private func findUrl(for social: String) -> URL? {
```

Comment:
I don't think this is the way to do this. I think you should loop through the socials array (`ForEach`) and check what kind of social it is. That solution is more scalable.

Reply:
I don't think scalability is something to consider when a single publication can only have max ~10 social links (some are stored in the backend even though we don't display them). I also want to maintain the order of socials and a `ForEach` would complicate that.

Reply:
I think 10 counts as scale in the sense that we don't want 30-40 lines of code that do more or less the same thing. When there are only 3 possible socials it's okay, but above that it'll become too long and just isn't clean imo. If you want to maintain order, you can sort with a custom predicate.